File: backend/app/models/enterprise.py
from sqlalchemy import Column, Integer, String, Float, DateTime, Enum as SQLEnum, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base
import enum


class EnterpriseStatus(str, enum.Enum):
    PROSPECT = "prospect"
    NEGOTIATION = "negotiation"
    PILOT = "pilot"
    ACTIVE = "active"
    INACTIVE = "inactive"


class EnterpriseCategory(str, enum.Enum):
    A = "A"  # Быстрые победы
    B = "B"  # Рабочие кейсы
    V = "V"  # Длинные проекты
    G = "G"  # Заморозка


class SalesStatus(str, enum.Enum):
    PLANNED = "planned"            # В планах
    CONTACT = "contact"            # Первый контакт
    NEGOTIATION = "negotiation"    # Переговоры
    CONTRACT = "contract"          # Договор
    LAUNCHED = "launched"          # Запущено


class Enterprise(Base):
    __tablename__ = "enterprises"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False, index=True)
    industry = Column(String(100))
    employee_count = Column(Integer)
    bank_penetration = Column(Float, default=0.0)  # % зарплатного проекта
    status = Column(SQLEnum(EnterpriseStatus), default=EnterpriseStatus.PROSPECT)

    # Pipeline fields
    category = Column(SQLEnum(EnterpriseCategory), default=EnterpriseCategory.V)
    score = Column(Integer, default=0)  # Скоринг-балл
    sales_status = Column(SQLEnum(SalesStatus), default=SalesStatus.PLANNED)

    inn = Column(String(12))  # ИНН предприятия (10 или 12 цифр)
    holding = Column(String(255))  # Холдинг/группа компаний
    locations = Column(String(1000))  # JSON string of locations
    contact_person = Column(String(255))
    contact_email = Column(String(255))
    contact_phone = Column(String(50))
    notes = Column(String(2000))
    manager = Column(String(255))  # Ответственный менеджер

    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    # Owner (for permission checks)
    created_by_id = Column(Integer, ForeignKey("users.id"), nullable=True)
    created_by = relationship("User", foreign_keys=[created_by_id])

    # Relationships
    kpp_contracts = relationship("KPPContract", back_populates="enterprise")
    interactions = relationship("Interaction", back_populates="enterprise", cascade="all, delete-orphan", order_by="desc(Interaction.date)")

File: frontend/src/constants/enterprise.ts
import type { ColumnConfig, Enterprise, NewInteraction } from '../types/enterprise'

// Default columns configuration
export const defaultColumns: ColumnConfig[] = [
  { id: 'category', label: 'Категория', visible: true, sortable: true },
  { id: 'name', label: 'Наименование', visible: true, sortable: true },
  { id: 'inn', label: 'ИНН', visible: true, sortable: true },
  { id: 'holding', label: 'Холдинг', visible: true, sortable: true },
  { id: 'industry', label: 'Отрасль', visible: true, sortable: true },
  { id: 'employee_count', label: 'Численность', visible: true, sortable: true },
  { id: 'manager', label: 'Менеджер', visible: true, sortable: true },
  { id: 'score', label: 'Балл', visible: true, sortable: true },
  { id: 'sales_status', label: 'Этап продаж', visible: true, sortable: true },
  { id: 'status', label: 'Статус', visible: true, sortable: true },
  { id: 'bank_penetration', label: 'Проникн. ЗП', visible: false, sortable: true },
  { id: 'locations', label: 'Площадки', visible: false, sortable: false },
  { id: 'contact_person', label: 'Контакт', visible: false, sortable: true },
  { id: 'contact_phone', label: 'Телефон', visible: false, sortable: false },
  { id: 'contact_email', label: 'Email', visible: false, sortable: false },
]

export const defaultEnterprise: Partial<Enterprise> = {
  name: '',
  industry: '',
  employee_count: 0,
  bank_penetration: 0,
  status: 'prospect',
  category: 'V',
  score: 0,
  sales_status: 'contact',
  inn: '',
  holding: '',
  locations: '',
  contact_person: '',
  contact_email: '',
  contact_phone: '',
  notes: '',
  manager: ''
}

export const defaultInteraction: NewInteraction = {
  interaction_type: 'call',
  description: '',
  result: '',
  created_by: ''
}

export const statusLabels: Record<string, string> = {
  prospect: 'Потенциал',
  negotiation: 'Переговоры',
  pilot: 'Пилот',
  active: 'Активный',
  inactive: 'Неактивный'
}

export const categoryLabels: Record<string, string> = {
  A: 'A — Быстрые победы',
  B: 'B — Рабочие кейсы',
  V: 'В — Длинные проекты',
  G: 'Г — Заморозка'
}

export const salesStatusLabels: Record<string, string> = {
  planned: 'В планах',
  contact: 'Первый контакт',
  negotiation: 'Переговоры',
  contract: 'Договор',
  launched: 'Запущено'
}

export const interactionTypeLabels: Record<string, string> = {
  call: 'Звонок',
  meeting: 'Встреча',
  email: 'Письмо',
  presentation: 'Презентация',
  contract: 'Работа с договором',
  other: 'Прочее'
}

export const interactionTypeIcons: Record<string, string> = {
  call: '📞',
  meeting: '🤝',
  email: '📧',
  presentation: '📊',
  contract: '📝',
  other: '📌'
}

export const salesStages = ['planned', 'contact', 'negotiation', 'contract', 'launched'] as const
